src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -140,6 +140,24 @@ internal static void ValidateScriptCompilationParameters(Compilation previousScr
             }
         }
 
+        internal static void CheckCompilationOptions(CompilationOptions options)
+        {
+            if (options == null)
+            {
+                return;
+            }
+
+            if (options.PublicSign)
+            {
+                var keyFile = options.CryptoKeyFile;
+                if (!string.IsNullOrWhiteSpace(options.CryptoKeyFile) &&
+                    !PathUtilities.IsAbsolute(options.CryptoKeyFile))
+                {
+                    throw new ArgumentException(CodeAnalysisResources.InvalidPublicKey + " " + CodeAnalysisResources.AbsolutePathExpected);
```

Comment:
The path should be substituted into the message resource (CodeAnalysisResources.InvalidPublicKey should have a {0} hole for it) rather than concatenated to it. Some languages might need a different order.